### PR TITLE
[Merged by Bors] - chore: move non `@[to_additive]` parts of `Algebra.Order.Monoid` and `Algebra.Order.Group` to a different file

### PR DESCRIPTION
**Pull request by Command-Master**
Following the discussion on #14598, move `LinearOrderedAddCommMonoidWithTop`, `LinearOrderedAddCommGroupWithTop`, and associated theorems to `Algebra.Order.AddGroupWithTop`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Command-Master** [comment]
!bench

File: Mathlib/Algebra/Order/Monoid/WithTop.lean [hunk 1]
```diff
@@ -100,374 +95,12 @@ protected theorem map_one {β} (f : α → β) : (1 : WithTop α).map f = (f 1 :
 #align with_top.map_one WithTop.map_one
 #align with_top.map_zero WithTop.map_zero
 
-instance zeroLEOneClass [Zero α] [LE α] [ZeroLEOneClass α] : ZeroLEOneClass (WithTop α) :=
+
```
Comment (YaelDillies) [inline Mathlib/Algebra/Order/Monoid/WithTop.lean:98]:
```suggestion
```

**Comment by YaelDillies** [comment]
Thanks! I think that's a pretty good split, and the new file is not too big.

maintainer merge

**Comment by YaelDillies** [comment]
Hmm, that performance in `CategoryTheory.Monoidal.Mon_` is a bit strange

**Comment by jcommelin** [comment]
But not too problematic, I think.

bors merge

**Pull request closed without merging**